src/js/utils/M3u8Parser.js

```diff
@@ -1,3 +1,25 @@
+function extractKeyInfo(keyLine) {
+    const keyInfo = keyLine.replace('#EXT-X-KEY:', '')
+    // All key value pairs are split with a ,
+    .split(',')
+    .reduce((result, keyValString) => {
+        // We only use the first equal sign per string
+        // Otherwise we might split something in an URL.
+        const firstEqualIndex = keyValString.indexOf('=');
+
+        // Retrieve key value pairs.
+        const key = keyValString.substring(0, firstEqualIndex);
+        const val = keyValString.substring(firstEqualIndex + 1);
+
+        // The values are displayed like: ""value"", so we remove the extra pair of "".
+        result[key] = val.replace(/"/g, '');
```

Comment:
This results in a linting error:
Assignment to property of function parameter 'result'  no-param-reassign

However, I think this is the correct way to do it so I would suggest disabling the param-reassign rule for this line.